weblate/addons/models.py

```diff
@@ -208,34 +208,42 @@ def handle_addon_error(addon, component):
         addon.disable()
 
 
-@receiver(vcs_pre_push)
-def pre_push(sender, component, **kwargs):
-    for addon in Addon.objects.filter_event(component, EVENT_PRE_PUSH):
-        component.log_debug("running pre_push add-on: %s", addon.name)
+def handle_addon_event(component, event_type, addon_method, description):
+    addons = Addon.objects.filter_event(component, event_type)
+
+    for addon in addons:
+        component.log_debug(f"running {event_type} add-on: {addon.name}")
         try:
-            with sentry_sdk.start_span(op="addon.pre_push", description=addon.name):
-                addon.addon.pre_push(component)
+            with sentry_sdk.start_span(
+                op=f"addon.{event_type}", description=addon.name
+            ):
+                addon_method(addon.addon, component)
         except DjangoDatabaseError:
             raise
         except Exception:
             handle_addon_error(addon, component)
         else:
-            component.log_debug("completed pre_push add-on: %s", addon.name)
+            component.log_debug(f"completed {event_type} add-on: {addon.name}")
+
+
+@receiver(vcs_pre_push)
+def pre_push(sender, component, **kwargs):
+    handle_addon_event(
+        component,
+        EVENT_PRE_PUSH,
+        lambda addon, comp, component=component: addon.pre_push(comp),
```

Comment:
Using lambdas here is not really a practical approach, as it makes the code hard to read, especially due to unused arguments for these.